internal/linter/check_reserved.go

```diff
@@ -0,0 +1,98 @@
+	"LONGTEXT", "LOOP", "LOW_PRIORITY", "MASTER_BIND", "MASTER_SSL_VERIFY_SERVER_CERT",
+	"MATCH", "MAXVALUE", "MEDIUMBLOB", "MEDIUMINT", "MEDIUMTEXT", "MIDDLEINT",
+	"MINUTE_MICROSECOND", "MINUTE_SECOND", "MOD", "MODIFIES", "NATURAL", "NOT",
+	"NO_WRITE_TO_BINLOG", "NTH_VALUE", "NTILE", "NULL", "NUMERIC", "OF", "ON",
+	"OPTIMIZE", "OPTIMIZER_COSTS", "OPTION", "OPTIONALLY", "OR", "ORDER", "OUT",
+	"OUTER", "OUTFILE", "OVER", "PARTITION", "PERCENT_RANK", "PRECISION", "PRIMARY",
+	"PROCEDURE", "PURGE", "RANGE", "RANK", "READ", "READS", "READ_WRITE", "REAL",
+	"RECURSIVE", "REFERENCES", "REGEXP", "RELEASE", "RENAME", "REPEAT", "REPLACE",
+	"REQUIRE", "RESIGNAL", "RESTRICT", "RETURN", "REVOKE", "RIGHT", "RLIKE", "ROW",
+	"ROWS", "ROW_NUMBER", "SCHEMA", "SCHEMAS", "SECOND_MICROSECOND", "SELECT",
+	"SENSITIVE", "SEPARATOR", "SET", "SHOW", "SIGNAL", "SMALLINT", "SPATIAL",
+	"SPECIFIC", "SQL", "SQLEXCEPTION", "SQLSTATE", "SQLWARNING", "SQL_BIG_RESULT",
+	"SQL_CALC_FOUND_ROWS", "SQL_SMALL_RESULT", "SSL", "STARTING", "STORED",
+	"STRAIGHT_JOIN", "SYSTEM", "TABLE", "TERMINATED", "THEN", "TINYBLOB", "TINYINT",
+	"TINYTEXT", "TO", "TRAILING", "TRIGGER", "TRUE", "UNDO", "UNION", "UNIQUE", "UNLOCK",
+	"UNSIGNED", "UPDATE", "USAGE", "USE", "USING", "UTC_DATE", "UTC_TIME", "UTC_TIMESTAMP",
+	"VALUES", "VARBINARY", "VARCHAR", "VARCHARACTER", "VARYING", "VIRTUAL", "WHEN", "WHERE",
+	"WHILE", "WINDOW", "WITH", "WRITE", "XOR", "YEAR_MONTH", "ZEROFILL",
+
+	// For MariaDB as at March 2023
+	// https://mariadb.com/kb/en/reserved-words/
+	// Adding just the extras, not in the MySQL list.
+	"CURRENT_ROLE", "DELETE_DOMAIN_ID", "DO_DOMAIN_IDS", "GENERAL",
+	"IGNORE_DOMAIN_IDS", "IGNORE_SERVER_IDS", "INTERSECTS", "MASTER_HEARTBEAT_PERIOD",
+	"OFFSET", "OVERLAPS", "PAGE_CHECKSUM", "PARSE_VCOL_EXPR", "POSITION", "REF_SYSTEM_ID", "RETURNING",
+	"SLOW", "STATS_AUTO_RECALC", "STATS_PERSISTENT", "STATS_SAMPLE_PAGES", "SYSTEM_USER",
+
+	// MariaDB Oracle-SQL Mode only reserved words.
+	// Uncomment to allow, currently disabled because it breaks other tests.
+	// "BODY", "ELSIF", "GOTO", "MINUS", "OTHERS", "PACKAGE", "RAISE", "ROWNUM", "ROWTYPE",
+	// "SYSDATE", "WITHOUT",
+
+	// For testing purposes, since if we try and run against a real MySQL/MariaDB server
+	// we might get a parse error before we get to this.
+	"SPECIAL_RESERVED_CHECKER_TEST_KEYWORD",
+}
+
+func reservedChecker(table *tengo.Table, _ string, _ *tengo.Schema, _ Options) *Note {
+	for _, reservedWord := range reservedWords {
+		if strings.EqualFold(table.Name, reservedWord) {
+			return &Note{
+				LineOffset: 0,
+				Summary:    "table name matches reserved word",
+				Message:    fmt.Sprintf("Table name %s matches a reserved word in a later MySQL/MariaDB release. For forward-compatibility consider renaming.", table.Name),
+			}
+		}
+		for _, col := range table.Columns {
+			if strings.EqualFold(col.Name, reservedWord) {
+				return &Note{
+					LineOffset: 0,
```

Comment:
This can use `FindColumnLineOffset(col, createStatement)` to attempt to find the correct line with the column's definition. It's not always accurate in some edge cases due to being based on a naive regex, but usually it's correct, and will be improved to be non-regex-based at some point in the future.